secretsmanager_resources.tf

```
@@ -29,6 +37,75 @@ resource "aws_secretsmanager_secret" "sidecar_created_certificate" {
   kms_key_id              = var.secrets_kms_arn
 }
 
+resource "aws_secretsmanager_secret" "sidecar_ca_certificate" {
+  name                    = "/cyral/sidecars/${var.sidecar_id}/ca-certificate"
+  description             = "CA certificate used by sidecar in case a custom CA certificate is not provided."
+  recovery_window_in_days = 0
+  kms_key_id              = var.secrets_kms_arn
+}
+
+resource "tls_private_key" "tls" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "tls_private_key" "ca" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "tls_self_signed_cert" "tls" {
+  private_key_pem   = tls_private_key.tls.private_key_pem
+  is_ca_certificate = false
+
+  subject {
+    country      = "US"
+    province     = "CA"
+    locality     = "Redwood City"
+    organization = "Cyral Inc."
+    common_name  = "sidecar.app.cyral.com"
+  }
+
+  validity_period_hours = 10 * 365 * 24
+
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+  ]
+}
+
+resource "tls_self_signed_cert" "ca" {
+  private_key_pem   = tls_private_key.tls.private_key_pem
+  is_ca_certificate = true
+
+  subject {
+    country      = "US"
+    province     = "CA"
+    locality     = "Redwood City"
+    organization = "Cyral Inc."
+  }
+
+  validity_period_hours = 10 * 365 * 24
```

```suggestion
  validity_period_hours = local.selfsigned_cert_validity_period_hours
```
Maybe we can also move this to a `locals { ... }` block, to avoid code duplication.